pkg/platform/api/buildplanner/model/buildexpression.go

```diff
@@ -56,9 +69,66 @@ type Requirement struct {
 type VersionRequirement map[string]string
 
 type BuildExpression struct {
-	expression   map[string]interface{}
-	solveNode    *map[string]interface{}
-	requirements []Requirement
+	Let *Let
+}
+
+type Let struct {
+	Assignments []*Var
+	In          *In
+}
+
+type Var struct {
+	Name  string
+	Value *Value
+}
+
+type Value struct {
+	Ap   *Ap
+	List *[]*Value
+	Str  *string
+	Null *Null
+
+	Assignment *Var
+	Object     *[]*Var
+	Ident      *string
+}
+
+type Null struct {
+	Null string
+}
+
+type Ap struct {
+	Name      string
+	Arguments []*Value
+}
+
+type In struct {
+	FuncCall *Ap
+	Name     *string
+}
+
+type context []string
```

Comment:
Perhaps place a documentation comment here? I won't know what this is or what it's for until I look for its use.